client/lib/push-notifications/index.js

```diff
@@ -95,8 +95,17 @@ PushNotifications.prototype.setState = function( state, callback ) {
 };
 
 PushNotifications.prototype.deleteSubscription = function() {
-	// @todo: delete the subscription
-	debug( 'Delete subscription' );
+	const deviceId = store.get( 'push-subscription-device-id' );
```

Comment:
We may just want to check to be sure that we receive a device id from the store before calling the endpoint.